benchmark/test_biocypher_query_generation.py

```diff
@@ -140,15 +150,300 @@ def test_end_to_end_query_generation(prompt_engine):
+            "GENE_EXPRESSED_IN_CELL_TYPE"
+        ).get("source")
+    )
+    score.append(
+        "CellType"
+        in prompt_engine.selected_relationship_labels.get(
+            "GENE_EXPRESSED_IN_CELL_TYPE"
+        ).get("target")
+    )
+
+    with open(FILE_PATH, "a") as f:
+        f.write(
+            f"{prompt_engine.model_name},relationships_multi,{calculate_test_score(score), prompt_engine.selected_relationships, prompt_engine.selected_relationship_labels}\n"
+        )
+
+
+# select properties DOES have issue
+
+
+def test_property_selection_multi_word(prompt_engine):
+    prompt_engine.question = "Which genes are expressed in fibroblasts?"
+    prompt_engine.selected_entities = ["Gene", "CellType"]
+    prompt_engine.selected_relationships = ["GeneExpressedInCellType"]
+    success = prompt_engine._select_properties()
+    assert success
+
+    score = []
+    score.append("CellType" in prompt_engine.selected_properties.keys())
+    if "CellType" in prompt_engine.selected_properties.keys():
+        # only run if CellType is actually a selected property
+        score.append(
+            "cell_type_name"
+            in prompt_engine.selected_properties.get("CellType")
+        )
+        # require a list of relevant properties
+        score.append(
+            isinstance(prompt_engine.selected_properties.get("CellType"), list)
+        )
+    else:
+        score.append(0)
+        score.append(0)
+
+    with open(FILE_PATH, "a") as f:
+        f.write(
+            f"{prompt_engine.model_name},properties_multi,{calculate_test_score(score), prompt_engine.selected_properties}\n"
+        )
+
+
+def test_query_generation_multi_word(prompt_engine):
+    query = prompt_engine._generate_query(
+        question="Which genes are expressed in fibroblasts?",
+        entities=["Gene", "CellType"],
+        relationships={
+            "GENE_EXPRESSED_IN_CELL_TYPE": {
+                "source": "Gene",
+                "target": "CellType",
+            },
+        },
+        properties={
+            "CellType": ["cell_type_name"],
+        },
+        query_language="Cypher",
+    )
+
+    score = []
+    score.append("MATCH" in query)
+    score.append("RETURN" in query)
+    score.append("Gene" in query)
+    score.append("CellType" in query)
+    score.append("fibroblast" in query)
+
+    # make sure direction is right
+    cypher_regex = r"MATCH \([a-zA-Z]*:Gene\)-\[[a-zA-Z]*:GENE_EXPRESSED_IN_CELL_TYPE\]->\([a-zA-Z]*:CellType.*|MATCH \([a-zA-Z]*:CellType.*<-\[[a-zA-Z]*:GENE_EXPRESSED_IN_CELL_TYPE\]-\([a-zA-Z]*:Gene\)"
+    score.append((re.search(cypher_regex, query) is not None))
+    # make sure conditioned
+    score.append("WHERE" in query or "{cell_type_name:" in query)
+
+    with open(FILE_PATH, "a") as f:
+        f.write(
+            f"{prompt_engine.model_name},cypher query_single,{calculate_test_score(score),query}\n"
+        )
+
+
+def test_end_to_end_query_generation_multi_word(prompt_engine):
```

Comment:
not sure we should do end-to-end tests for all scenarios, maybe the simple one is enough. in theory, this should be covered by the step-wise tests, but maybe keeping one of them is fine. many of those could needlessly increase cost/runtime.

Reply:
Only kept single word end-to-end test.